src/box/lua/config/applier/credentials.lua

```diff
@@ -834,6 +840,92 @@ end
 
 -- }}} Create users
 
+-- {{{ Drop absent users and roles
+
+local function drop_users_not_in_config(user_map)
+    local to_drop = {}
+    for _, tuple in box.space._user:pairs() do
+        if tuple.type == 'user' then
+            local origins = user_origins_from_tuple(tuple)
+            if origins[CONFIG_ORIGIN] and
+               (user_map or {})[tuple.name] == nil then
+                table.insert(to_drop, tuple.name)
+            end
+        end
+    end
+    for _, name in ipairs(to_drop) do
+        log.verbose('credentials.apply: drop user %q', name)
+        box.schema.user.drop(name, {_origin = CONFIG_ORIGIN})
+    end
+end
+
+local function drop_roles_not_in_config(role_map)
+    local to_drop = {}
+    for _, tuple in box.space._user:pairs() do
+        if tuple.type == 'role' then
+            local origins = user_origins_from_tuple(tuple)
+            if origins[CONFIG_ORIGIN] and
+               (role_map or {})[tuple.name] == nil then
+                table.insert(to_drop, tuple.name)
+            end
+        end
+    end
+    for _, name in ipairs(to_drop) do
+        log.verbose('credentials.apply: drop role %q', name)
+        box.schema.role.drop(name, {_origin = CONFIG_ORIGIN})
+    end
+end
+
+-- }}} Drop absent users and roles
+
+-- {{{ Manual cleanup alerts
+
+local function has_unmanaged_subjects()
+    for _, tuple in box.space._user:pairs() do
+        local uid = tuple.id
+        if uid > box.schema.SYSTEM_USER_ID_MAX and
+           uid ~= box.schema.SUPER_ROLE_ID then
+            local origins = user_origins_from_tuple(tuple)
+            if origins[DEFAULT_ORIGIN] then
+                return true
+            end
+        end
+    end
+    return false
+end
+
+local function refresh_unmanaged_subjects_alert()
+    if has_unmanaged_subjects() then
+        local orphan_users_alert_key = 'credentials_orphan_users'
+
+        local script_url =
+            'https://raw.githubusercontent.com/tarantool/tarantool/' ..
+            'refs/heads/master/tools/find-orphan-users.lua'
+
+        local msg = textutils.dedent(([[
+            Found users/roles authored from Lua and not managed by
+            the YAML configuration.
+
+            Use the helper script `find-orphan-users.lua` to list
+            duplicates/orphans and print commands for transferring
+            ownership to the YAML configuration.
+
+            dofile('/path/to/find-orphan-users.lua')
+            -- or fetch the latest version:
+            -- %s
+        ]]):format(script_url)) .. '\n'
+
+        config._aboard:set({
+            type = 'warn',
+            message = msg,
+        }, {key = orphan_users_alert_key})
+
+        config._aboard:drop(orphan_users_alert_key)
```

Comment:
What is the point to set the warning and then immediatelly drop it? It looks like the `else` keyword is forgotten in between.